manifests/init.pp

```diff
@@ -74,17 +74,22 @@
       $_tz = split($timezone, '/')
       $area = $_tz[0]
       $zone = $_tz[1]
-      exec { 'update_debconf area':
-        command => "echo tzdata tzdata/Areas select ${area} | debconf-set-selections",
-        unless  => "debconf-get-selections |grep -q -E \"^tzdata\\s+tzdata/Areas\\s+select\\s+${area}\"",
-        path    => $facts['path'],
-      }
-      exec { 'update_debconf zone':
-        command => "echo tzdata tzdata/Zones/${area} select ${timezone} | debconf-set-selections",
-        unless  => "debconf-get-selections |grep -E \"^tzdata\\s+tzdata/Zones/${area}\\s+select\\s+${zone}\"",
-        path    => $facts['path'],
+
+      debconf {
+        'tzdata/Areas':
+          package => 'tzdata',
+          item    => 'tzdata/Areas',
+          type    => 'select',
+          value   => $area;
+        "tzdata/Zones/${area}":
+          package => 'tzdata',
+          item    => "tzdata/Zones/${area}",
+          type    => 'select',
+          value   => $zone;
       }
+      -> Package[$package]
     }
+
     package { $package:
```

Comment:
The newer saz/timezone has a block that handles setting the hardware clock (see below).  Why didn't we pick that up?
 if $ensure == 'present' and $hwutc != undef {
    $hwclock_cmd = lookup('timezone::hwclock_cmd', Optional[String], 'first', undef)
    $hwclock_check_enabled_cmd = lookup('timezone::check_hwclock_enabled_cmd', Optional[String], 'first', undef)
    $hwclock_check_disabled_cmd = lookup('timezone::check_hwclock_disabled_cmd', Optional[String], 'first', undef)

    if $hwclock_cmd != '' and $hwclock_cmd != undef {
      if ! $hwutc {
        $hwclock_unless = $hwclock_check_enabled_cmd
      } else {
        $hwclock_unless = $hwclock_check_disabled_cmd
      }
      exec { 'set_hwclock':
        command => sprintf($hwclock_cmd, (! $hwutc)),
        unless  => $hwclock_unless,
        path    => '/usr/bin:/usr/sbin:/bin:/sbin',
      }
    }
  }